lib/storage/providers/LocalForage.js

```diff
@@ -57,12 +62,37 @@ const provider = {
      * @return {Promise<void>}
      */
     multiMerge(pairs) {
-        const tasks = _.map(pairs, ([key, value]) => this.setItemQueue.push({key, value, shouldMerge: true}));
+        const tasks = _.map(pairs, ([key, value, currentValue]) => {
+            if (mergeQueue[key]) {
+                mergeQueue[key].push(value);
+                return Promise.resolve();
```

Comment:
Another place where the method resolve before the write is over 

This might swallow errors, like prevents us from running the storage eviction logic to cleanup space 

There should be a way to wait and resolve after `mergeQueue[key]` is done